File: backend/app/schemas/patch_log.py
"""Patch Log DTO"""
from __future__ import annotations

from datetime import datetime
from typing import Any, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class PatchLogResponse(BaseModel):
    id: UUID
    scene_version_id: UUID
    created_by: Optional[UUID] = None
    patch_type: str
    target_type: str
    target_id: UUID
    patch_json: dict[str, Any] = Field(default_factory=dict)
    created_at: datetime

File: frontend/src/types/patch-log.ts
import type { ISODateString, UUID } from './common';

export interface PatchLog {
  id: UUID;
  scene_version_id: UUID;
  created_by: UUID | null;
  patch_type: string;
  target_type: 'room' | 'wall' | 'opening' | 'object' | string;
  target_id: UUID;
  patch_json: Record<string, unknown>;
  created_at: ISODateString;
}
